app/components/app_consent_component.rb

```diff
@@ -55,12 +55,20 @@ def contact_parent_or_guardian_link(consents)
 
   def consents_grouped_by_parent
     @consents_grouped_by_parent ||=
-      @patient_session.consents.group_by do |consent|
-        response = consent.human_enum_name(:response).capitalize
-        summary = "#{response} by #{consent.name}"
-        summary +=
-          " (#{consent.who_responded})" unless consent.via_self_consent?
-        summary
+      @patient_session.consents.group_by { |consent| summary_for(consent) }
+  end
+
+  private
+
+  def summary_for(consent)
+    summary =
+      if consent.respond_to?(:response_not_provided?) &&
```

Comment:
Why do we need to check it responds to `response_not_provided?`?

Reply:
because `consent` here can be either a `Consent` or a `ConsentForm`. Perhaps we shouldn't be using the same component for displaying both.